src/features2d/surf.rs

```diff
@@ -103,11 +82,18 @@ impl Into<SURF> for SURFBuilder {
 
 impl Feature2D for SURF {
     fn detect_and_compute(&self, image: &Mat, mask: &Mat) -> (Vec<KeyPoint>, Mat) {
-        let mut keypoints = CVec::<KeyPoint>::default();
-        let descriptors = CMat::new();
         unsafe {
-            cv_surf_detect_and_compute(self.value, image.inner, mask.inner, &mut keypoints, descriptors, false);
+            let mut keypoints: native::cvsys_CVec<native::cvsys_KeyPoint> = std::mem::zeroed();
```

Comment:
+1 Nullptr/alloc question here

Reply:
This code is not ideal (I shouldn't be using `std::mem::zeroed`), but on the C++ side the pointer is filled in and when the `CVec` is dropped the C++ side cleans up the memory if the pointer is set. In the future, or possibly now, we should create a safe `new` method. This was just a quick hack.